### DISCUSSION TITLE
IoT with Bluetooth DMM using MQTT / Websockets / etc.

### DISCUSSION THREAD

**Discussion by ludwich66** [webspiderteam/Bluetooth-DMM-For-Windows#10] [Ideas]
My idea is to add an interface to the IoT world to the B-DMMfW software.
Here I would suggest a "standard" data exchange like MQTT or websockets.
There are numerous applications that can use and evaluate this data.
As an application scenario, there are certainly a number of topics that require long-term logging, but do not require permanent installation of a measuring device.

**For example:**

- Battery charging and discharging cycles
- Humidity logging (water damage etc.)
- temperature monitoring
- check wallboxes 

In the attachment a model representation.

In order not to inflate the application unnecessarily, a QOS (Quality of Service) 0 would be sufficient (fire&forgot) from my point of view. The payload should have at least the scope of the current csv export.

I look forward to a lively discussion and more ideas.

<img width="1138" alt="image" src="https://user-images.githubusercontent.com/12202733/181717859-5a717de7-2895-44c3-a720-ef658eb66676.png">

Grettings ludwich

**Comment by webspiderteam** [comment]
I can add MQTT client in app but without Compatible Broker  its not usable I think. I can send data every second. also client settings can be done easily. If you really wish me to add client its ok. But Server side is huge project.

**Comment by ludwich66** [comment]
First good Morning!

Client Side only is Ok in my Opinion

ludwich

**Comment by webspiderteam** [comment]
Ok then I will start to work on it.

**Comment by webspiderteam** [comment]
Release is Done :)

**Comment by ludwich66** [comment]
New Idea to Dataprotocol
These are all just ideas up for discussion.

You want to keep the amount of data small. 
In order to keep the datagram small, one could make it easily configurable (gold-version!).
Click together the key value pairs you need.
Time : Timstamp
Device : device name
Current : Kind_of_Current
Value_Str : 8888 and auto
Unit_Str: mA, A, µ, n, p, k, M
Value SI: Float (10-9 to 10+9)
Unit SI: V, A, Ω, Hz, F
Icon: Buzzer, H.V. Arrow, Max, Min, %, Inrush, Peek

JSON KEY | JSON VALUE |  
-- | -- | --
Time: | "01.08.2022 12:10:34" |  
Device: | "Bluetooth DMM" | your-own-name
Value_SI: | 0,000.008.888 - 8.888.000.000.000 | FLOAT, 6000/ 19000 Dig
Unit_SI: | V, A, F, Ω, Hz , °C/°F|  
Value: | 0000, 0.000, 00.00, 000.0, Auto, .OL |  +Strings, Auto, .OL. ...
Unit: | V, mV | U
  | µA, mA, A | I
  | nF, µF, mF | L
  | Ω, kΩ, MΩ | R
  | Hz, kHz, MHz | Frequency
  ||°C, °F | Temp
Current: | „ “, AC, DC |  
Symbol: | BUZ, HOLD, INRUSH, PEEK, MIN, MAX, H.V., REL |  

OR 

Simpel add  **Value_SI and Unit_SI** 
I then test whether the amount of data causes problems in transport

**Comment by webspiderteam** [comment]
working on it

**Reply by ludwich66** [reply]
thumbs up!!

First draft of Wiki is online

**Comment by webspiderteam** [comment]
Oh It looks nice :) thank you 🥇

**Comment by ks00x** [comment]
yeah!  Got it to work with the MQTT server and Node-Red on my Raspi. Pretty cool - thanks! I entered the ip addr of my raspi into the MQTT server field and subscribed to BT_DMM/Values  with the nodered mqtt in node
